xrpl/client/websocket/client.go

```diff
@@ -0,0 +1,148 @@
+package websocket
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"sync/atomic"
+
+	"github.com/Peersyst/xrpl-go/xrpl/client"
+	"github.com/Peersyst/xrpl-go/xrpl/model/transactions"
+	"github.com/Peersyst/xrpl-go/xrpl/model/transactions/types"
+	"github.com/gorilla/websocket"
+)
+
+const (
+	DEFAULT_FEE_CUSHION float32 = 1.2
+	DEFAULT_MAX_FEE_XRP float32 = 2
+)
+
+var _ client.Client = (*WebsocketClient)(nil)
+
+var ErrIncorrectId = errors.New("incorrect id")
+
+type WebsocketConfig struct {
+	URL string
+}
+
+type WebsocketClient struct {
+	cfg       *WebsocketConfig
+	idCounter atomic.Uint32
+
+	NetworkId  uint32
+	FeeCushion float32
+	MaxFeeXRP  float32
+}
+
+// Creates a new websocket client with cfg.
+// This client will open and close a websocket connection for each request.
+func NewWebsocketClient(cfg *WebsocketConfig) *WebsocketClient {
+	return &WebsocketClient{
+		cfg:        cfg,
+		FeeCushion: DEFAULT_FEE_CUSHION,
+		MaxFeeXRP:  DEFAULT_MAX_FEE_XRP,
+	}
+}
+
+func NewClient(cfg *WebsocketConfig) *client.XRPLClient {
+	wcl := &WebsocketClient{
+		cfg: cfg,
```

Comment:
Is it possible to configure the FeeCushion and MaxFeeXRP in this `NewClient` function? I can only see the `cfg` being passed.